sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpProducer.cs

```diff
@@ -443,6 +529,7 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
         /// </remarks>
```

Comment:
Generally, I'm really not a fan of checked exceptions - either by the type system or via comments.  It tends to either end up being "this can throw an exception" or trying to enumerate every possible exception that could occur and keep that up to date.  I'm not sure that either is helpful to callers.

I do see value in calling out special cases for members where there are specific, known error scenarios that we want callers to consider handling.   In this case, the type is `internal` and the exception would really be "we could throw anything here", so I'm going to skip out.

_(not sure that you meant this for CloseAsync or as a companion to the NotImplementedException deal)_